target_chains/sui/contracts/sources/pyth.move

```diff
@@ -646,17 +770,373 @@ module pyth::pyth_tests{
+        assert!(vector::length<ID>(&shared_ids)==3, 0);
+        assert!(vector::length<ID>(&created_ids)==3, 0);
+
+        let price_info_object_1 = take_shared<PriceInfoObject>(&scenario);
+        let price_info_object_2 = take_shared<PriceInfoObject>(&scenario);
+        let price_info_object_3 = take_shared<PriceInfoObject>(&scenario);
+
+        // Create authenticated price infos
+        verified_vaa = get_verified_vaa_from_accumulator_message(&worm_state, TEST_ACCUMULATOR_3_MSGS, &clock);
+        let auth_price_infos = pyth::create_authenticated_price_infos_using_accumulator(
+            &pyth_state,
+            TEST_ACCUMULATOR_3_MSGS,
+            verified_vaa,
+            &clock
+        );
+
+        let coins2 = coin::split(&mut coins, 1000, ctx(&mut scenario));
+        let coins3 = coin::split(&mut coins, 1000, ctx(&mut scenario));
+
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+
+        // Update price feeds
+        auth_price_infos = update_single_price_feed(
+            &mut pyth_state,
+            auth_price_infos,
+            &mut price_info_object_1,
+            coins,
+            &clock
+        );
+
+        auth_price_infos = update_single_price_feed(
+            &mut pyth_state,
+            auth_price_infos,
+            &mut price_info_object_2,
+            coins2,
+            &clock
+        );
+
+        auth_price_infos = update_single_price_feed(
+            &mut pyth_state,
+            auth_price_infos,
+            &mut price_info_object_3,
+            coins3,
+            &clock
+        );
+
+        // assert price feeds are as expected
+        let expected_price_infos = accumulator_test_3_to_price_info(0 /*offset argument*/);
+
+        let price_info_1 = price_info::get_price_info_from_price_info_object(&price_info_object_1);
+        assert!(price_feeds_equal(&price_info_1, vector::borrow(&expected_price_infos, 0)), 0);
+
+        let price_info_2 = price_info::get_price_info_from_price_info_object(&price_info_object_2);
+        assert!(price_feeds_equal(&price_info_2, vector::borrow(&expected_price_infos, 1)), 0);
+
+        let price_info_3 = price_info::get_price_info_from_price_info_object(&price_info_object_3);
+        assert!(price_feeds_equal(&price_info_3, vector::borrow(&expected_price_infos, 2)), 0);
+
+
+        // clean up test scenario
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+        authenticated_vector::destroy<PriceInfo>(auth_price_infos);
+
+        return_shared(pyth_state);
+        return_shared(worm_state);
+        return_shared(price_info_object_1);
+        return_shared(price_info_object_2);
+        return_shared(price_info_object_3);
+
+        clock::destroy_for_testing(clock);
+        test_scenario::end(scenario);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = pyth::pyth::E_INSUFFICIENT_FEE)]
+    fun test_create_and_update_price_feeds_insufficient_fee() {
```

Comment:
nitpik: It's better to use accumulator message here (we might remove batch attestation soon)